lib/mix/lib/mix/scm/git.ex

```diff
@@ -192,6 +192,15 @@ defmodule Mix.SCM.Git do
     end
   end
 
+  defp git_repos_match?(repo_a, repo_b) do
+    normalize_github_repo(repo_a) == normalize_github_repo(repo_b)
+  end
+
+  defp normalize_github_repo(repo) do
+    normalized = String.replace(repo, "git://git@github.com:", "https://github.com/")
```

Comment:
Don't use a string replace, do pattern matching instead:

    defp normalize_github_repo("git://github.com/" <>  rest), do: "https://github.com/..."

Also, do not rewrite `git://git@github.com:` because in this case they are very different (one can access  private repos, the other cannot).